_posts/2021-09-26-new-multiplayer-experience.md

```diff
@@ -0,0 +1,185 @@
+---
+title: New Multiplayer Experience
+author: TrueBrain
+---
+
+With 12.0 in feature-freeze, it is time to talk about what is in 12.0, and why we release early.
+
+In summary: 12.0 makes setting up multiplayer games painless.
+You no longer need to configure anything in your home network.
+Playing together are now just four simple steps:
+1) You just start a server.
+2) You set the server to invite-only/public.
+3) You share your invite-code with your friend.
+4) Your friend joins your server based on the invite-code.
+
+There is no need to setup port-forwarding, or anything like that.
+The above will just work, for everyone, as long as you have a working Internet connection.
+
+Depending on your connection, it costs us extra in infrastructure costs, but we hope to cover that with donations.
+If our regular donations (thank you so much for those!) run out, we will let you know, and run some kind of fundraiser.
+In a sense, I guess, this post is also a fundraiser ;)
+
+As you might understand, this changes the multiplayer experience drastically, and we felt that waiting 6 more months to deliver that to you is not fair.
+In result: we are releasing 12.0 early!
+
+In this post I want to explain in a bit more details how we got here and how it works.
+
+<!-- more -->
+
+After our release on Steam, you guys wrote a lot of positive reviews.
+Over 95% of them are positive!
+I cannot express how happy that makes me: so many of you still enjoy the game, and this number alone makes it worth continue to work on OpenTTD.
+
+But that leaves the 5% that did not like the game.
+Although some reviews are just .. weird, a lot of them had clearly written out what they were missing.
+
+The number 1, without a doubt is: no tutorial.
+Okay, we see you, and even agree.
+Just there is no simple solution for that .. we have some ideas, but it will take a while before that takes shape.
+
+The number 2 was much more interesting to me: I cannot play with my friend.
+I even personally notice that, that I often don't play OpenTTD as I don't fancy playing alone.
+And setting up port-forwarding is, even for me, not trivial.
+So, I don't.
+
+This made me wonder: how hard is it to solve this problem in 2021?
+Back in 2007 (when this network implementation was made), there was no real solution, but we are 14 years later .. so, what can we do?
+
+Initially we looked into how Steam does it: SDR (Steam Datagram Relay).
+Basically, games using the Steam Multiplayer are likely to use this, and it means all traffic is routed via Steam.
+This means that if you can download Steam, you can play together.
+This works really well for a lot of people.
+And a lot of games make use of this.
+
+The problem for us is two-fold:
+using something like this would mean that all the non-Steam players cannot enjoy it, and we still are not sure how to integrate with the Steam SDK.
```

Comment:
```suggestion
Using something like this would mean that all the non-Steam players cannot enjoy it, and we still are not sure, from a license-standpoint, how to integrate with the Steam SDK.
```